src/classes/FeatureParameterService_TEST.cls

```diff
@@ -0,0 +1,96 @@
+    private static void getInstanceNew() {
+
+        Test.startTest();
+            FeatureParameterService featureParamServiceInstance = FeatureParameterService.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(FeatureParameterService.instance, featureParamServiceInstance, 'Instance of service class returned should match static instance variable.');
+        System.assertEquals(true, featureParamServiceInstance != null, 'Instance of service class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getInstance method returns the existing instance of the 
+    * FeatureParameterService class when one already exists.
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceExisting() {
+
+        Test.startTest();
+            FeatureParameterService featureParamServiceInstance1 = FeatureParameterService.getInstance();
+            FeatureParameterService featureParamServiceInstance2 = FeatureParameterService.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(FeatureParameterService.instance, featureParamServiceInstance1, 'Instance of service class returned should match static instance variable.');
+        System.assertEquals(featureParamServiceInstance1, featureParamServiceInstance2, 'Subsequent retrievals of service class instance should return existing instance.');
+        System.assertEquals(true, featureParamServiceInstance1 != null, 'Instance of service class should not be null.');
+    }
+
+    /****************************************************************************************************************************
+    * @description Test method to verify that the locateFeatureParameterMapper method returns instance of FeatureParameterMapper
+    ****************************************************************************************************************************/
+    @isTest
+    private static void locateFeatureParameterMapper() {
+
+        Test.startTest();
+            FeatureParameterMapper featureParamMapper = FeatureParameterService.getInstance().locateFeatureParameterMapper();
+        Test.stopTest();
+
+        System.assertEquals(FeatureParameterMapper.getInstance(), featureParamMapper, 'Should return the instance of FeatureParameterMapper');
+
+    }    
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getFeatureParameterModel method returns FeatureParameterModel
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getFeatureParameterModel() {
```

Comment:
Let's change this to use a stubbed FeatureParameterMapper that always returns an expected value, which in this case would be a new instance of a FeatureParameterDateModel. We can then check against that to ensure an encapsulated passthrough.